### PR TITLE
Cleaning orders

### PR DESCRIPTION
**Pull request by brittcyr**

### CODE REVIEW

File: programs/manifest/src/program/processor/global_clean.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    } = global_clean_context;
+
+    let global_trade_accounts: GlobalTradeAccounts = GlobalTradeAccounts {
+        mint_opt: None,
+        global: global.clone(),
+        global_vault_opt: None,
+        market_vault_opt: None,
+        token_program_opt: None,
+        system_program: Some(system_program),
+        trader: payer,
```
Comment (mschneider) [inline programs/manifest/src/program/processor/global_clean.rs:50]: is this line what enables the indirect transfer from the deposit an user made for his order that is not being cleaned to the person invoking global clean?

Comment (brittcyr) [reply programs/manifest/src/program/processor/global_clean.rs:50]: Kinda. There is only one account info that gives or receives gas, the signer. We don't even have the other persons account info. Can document and rename to something else if you think of something better

Comment (brittcyr) [reply programs/manifest/src/program/processor/global_clean.rs:50]: This passes in the account info and then place and cancel credit or debit it with gas payments

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs:50]: yeah the main confusion here comes from the re-use of the trader field, which intuitively would be the person that placed the order with the payer identity which really could be someone who never trades

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs:50]: in general it's not ideal if a method has 10+ arguments and a change in one of them (especially hidden in a struct) can cause critical bugs

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs:50]: i would favor a more explicit code style

File: programs/manifest/src/validation/loaders.rs [hunk 2]
```diff
@@ -370,22 +370,22 @@ impl<'a, 'info> SwapContext<'a, 'info> {
...
-                    quote_vault.clone()
+                    Some(quote_vault.clone())
                 },
-                token_program: if index == 0 {
-                    token_program_base.clone()
+                token_program_opt: if index == 0 {
+                    Some(token_program_base.clone())
                 } else {
-                    token_program_quote.clone()
+                    Some(token_program_quote.clone())
```
[the remark this answers is not in the document]
Comment (brittcyr) [reply programs/manifest/src/validation/loaders.rs:388]: here fine?
https://github.com/CKS-Systems/manifest/blob/main/programs/manifest/tests/cases/loaders.rs

or you need unit test in this file?

File: programs/manifest/src/program/processor/global_clean.rs (hunk 1 above)
Comment (brittcyr) [reply programs/manifest/src/program/processor/global_clean.rs:50]: or do you want to throw away the "give me a ref to everything I could need for global" struct altogether

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs]: gas_receiver_opt & gas_payer_opt definitely an improvement. i thought to first split the method and then have another look at the struct how it's used post-merge to see what's used where, so we can make a decision how to break it up

Comment (brittcyr) [reply programs/manifest/src/program/processor/global_clean.rs]: which method? this everything struct is passed through for both place and cancel

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs]: remove_from_global

Comment (mschneider) [reply programs/manifest/src/program/processor/global_clean.rs]: are you still planning to split out gas_receiver into it's own account?

**Review by mschneider — APPROVED** [review]

**Pull request merged**